### ISSUE TITLE
test_xfrm* from the systemd-networkd-testsuite.py fail on the latest Arch

### ISSUE THREAD

**Issue by mrc0mmand** [systemd/systemd#22329]
**systemd version the issue has been seen with**
 > recent main

**Used distribution**
 > Arch Linux

**Linux kernel version used** (`uname -a`)

 > `5.16.4-arch1-1`

Not sure if it's the kernel update (5.16.2-arch1-1 -> 5.16.4-arch1-1) or something else, but `test_xfrm` and `test_xfrm_independent` keep failing when CI attempts to upgrade the Arch Linux image:

```
======================================================================
ERROR: test_xfrm (__main__.NetworkdNetDevTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/./test/test-network/systemd-networkd-tests.py", line 674, in wait_online
    check_output(*args, env=env)
  File "/build/./test/test-network/systemd-networkd-tests.py", line 59, in check_output
    return subprocess.check_output(command, universal_newlines=True, **kwargs).rstrip()
  File "/usr/lib/python3.10/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/lib/systemd/systemd-networkd-wait-online', '--timeout=20s', '--interface=xfrm99:degraded', '--interface=dummy98:degraded']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/build/./test/test-network/systemd-networkd-tests.py", line 1803, in test_xfrm
    self.wait_online(['xfrm99:degraded', 'dummy98:degraded'])
  File "/build/./test/test-network/systemd-networkd-tests.py", line 677, in wait_online
    output = check_output(*networkctl_cmd, '-n', '0', 'status', link.split(':')[0], env=env)
  File "/build/./test/test-network/systemd-networkd-tests.py", line 59, in check_output
    return subprocess.check_output(command, universal_newlines=True, **kwargs).rstrip()
  File "/usr/lib/python3.10/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/networkctl', '-n', '0', 'status', 'xfrm99']' returned non-zero exit status 1.

======================================================================
ERROR: test_xfrm_independent (__main__.NetworkdNetDevTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/./test/test-network/systemd-networkd-tests.py", line 674, in wait_online
    check_output(*args, env=env)
  File "/build/./test/test-network/systemd-networkd-tests.py", line 59, in check_output
    return subprocess.check_output(command, universal_newlines=True, **kwargs).rstrip()
  File "/usr/lib/python3.10/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/lib/systemd/systemd-networkd-wait-online', '--timeout=20s', '--interface=xfrm99:degraded']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/build/./test/test-network/systemd-networkd-tests.py", line 1813, in test_xfrm_independent
    self.wait_online(['xfrm99:degraded'])
  File "/build/./test/test-network/systemd-networkd-tests.py", line 677, in wait_online
    output = check_output(*networkctl_cmd, '-n', '0', 'status', link.split(':')[0], env=env)
  File "/build/./test/test-network/systemd-networkd-tests.py", line 59, in check_output
    return subprocess.check_output(command, universal_newlines=True, **kwargs).rstrip()
  File "/usr/lib/python3.10/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/networkctl', '-n', '0', 'status', 'xfrm99']' returned non-zero exit status 1.

----------------------------------------------------------------------
Ran 159 tests in 2361.178s

FAILED (errors=2, skipped=1, expected failures=3)
```

Relevant lines from the journal:
```
systemd-networkd[364474]: dummy98: udev initialized link
systemd-networkd[364474]: dummy98: State changed: pending -> initialized
systemd-networkd[364474]: dummy98: Link state is up-to-date
systemd-networkd[364474]: dummy98: found matching network '/run/systemd/network/xfrm.network'.
systemd-networkd[364474]: dummy98: State changed: initialized -> configuring
systemd-networkd[364474]: Setting '/proc/sys/net/ipv6/conf/dummy98/disable_ipv6' to '0'
systemd-networkd[364474]: No change in value '0', suppressing write
systemd-networkd[364474]: Setting '/proc/sys/net/ipv6/conf/dummy98/use_tempaddr' to '0'
systemd-networkd[364474]: No change in value '0', suppressing write
systemd-networkd[364474]: Setting '/proc/sys/net/ipv6/conf/dummy98/accept_ra' to '0'
systemd-networkd[364474]: Setting '/proc/sys/net/ipv6/conf/dummy98/proxy_ndp' to '0'
systemd-networkd[364474]: No change in value '0', suppressing write
systemd-networkd[364474]: Setting '/proc/sys/net/ipv4/conf/dummy98/promote_secondaries' to '1'
systemd-networkd[364474]: No change in value '1', suppressing write
systemd-networkd[364474]: dummy98: Requested to set interface group
systemd-networkd[364474]: dummy98: Requested to set IPv6LL address generation mode
systemd-networkd[364474]: dummy98: Requested to set master interface
systemd-networkd[364474]: dummy98: Requested stacked netdev 'xfrm99'
systemd-networkd[364474]: dummy98: Requested to activate link
systemd-networkd[364474]: dummy98: link_check_ready(): link layer is configuring.
systemd-networkd[364474]: dummy98: link_check_ready(): link layer is configuring.
systemd-networkd[364474]: dummy98: link_check_ready(): link layer is configuring.
systemd-networkd[364474]: dummy98: link_check_ready(): link layer is configuring.
systemd-networkd[364474]: dummy98: link_check_ready(): link layer is configuring.
systemd-networkd[364474]: dummy98: link_check_ready(): link layer is configuring.
systemd-networkd[364474]: dummy98: link_check_ready(): link layer is configuring.
systemd-networkd[364474]: dummy98: link_check_ready(): link layer is configuring.
systemd-networkd[364474]: dummy98: link_check_ready(): link layer is configuring.
systemd-networkd[364474]: dummy98: Setting interface group
systemd-networkd[364474]: dummy98: Setting IPv6LL address generation mode
systemd-networkd[364474]: dummy98: Setting master interface
systemd-networkd[364474]: dummy98: interface group set.
systemd-networkd[364474]: dummy98: IPv6LL address generation mode set.
systemd-networkd[364474]: dummy98: master interface set.
systemd-networkd[364474]: dummy98: link_check_ready(): link is not activated.
systemd-networkd[364474]: xfrm99: Creating
systemd-networkd[364474]: dummy98: Bringing link up
systemd-networkd[364474]: dummy98: Could not create stacked netdev: if_id must be non zero. Invalid argument
systemd-networkd[364474]: dummy98: Failed
systemd-networkd[364474]: dummy98: State changed: configuring -> failed
```

```
systemd-networkd[364521]: xfrm99: loaded xfrm
systemd-networkd[364521]: xfrm99: Creating
...
systemd-networkd[364521]: xfrm99: netdev could not be created: Invalid argument
```

Full log: [systemd-networkd-tests.py_FAIL.log](https://github.com/systemd/systemd/files/7974446/systemd-networkd-tests.py_FAIL.log)

I restarted the image upgrade just in case this one was a fluke, will post an update tomorrow.

/cc @yuwata

**Comment by yuwata** [comment]
5.16.3 has several patches for xfrm. E.g. `if_id` related message is introduced by https://git.kernel.org/pub/scm/linux/kernel/git/stable/stable-queue.git/tree/releases/5.16.3/xfrm-interface-with-if_id-0-should-return-error.patch.

**Comment by yuwata** [comment]
Fix is waiting in #22331.
